x-pack/plugins/observability/server/services/slo/tasks/summary_cleanup_task.ts

```diff
@@ -0,0 +1,208 @@
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { SortResults } from '@elastic/elasticsearch/lib/api/types';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../../assets/constants';
+import { StoredSLO } from '../../../domain/models';
+import { SO_SLO_TYPE } from '../../../saved_objects';
+
+export const TASK_TYPE = 'SLO:SUMMARIES-CLEANUP-TASK';
+
+function findMissingItems(soIds: string[], sloSummaryIds: string[]) {
+  const missingSummaryIds = sloSummaryIds.filter((item) => !soIds.includes(item));
+  const missingSOIds = soIds.filter((item) => !sloSummaryIds.includes(item));
+
+  return {
+    missingSummaryIds,
+    missingSOIds,
+  };
+}
+
+const SEPARATOR = '__V__';
+
+export const getDeleteQueryFilter = (sloSummaryIdsToDelete: string[]) => {
+  return sloSummaryIdsToDelete.map((id) => {
+    const [sloId, revision] = id.split(SEPARATOR);
+    return {
+      bool: {
+        must: [
+          {
+            term: {
+              'slo.id': sloId,
+            },
+          },
+          {
+            term: {
+              'slo.revision': Number(revision),
+            },
+          },
+        ],
+      },
+    };
+  });
+};
+
+export class SloSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClientContract;
+  private esClient?: ElasticsearchClient;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger) {
+    this.logger = logger;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '3m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask();
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  runTask = async () => {
+    const runAt = new Date().toISOString();
+
+    if (this.soClient && this.esClient) {
+      const finder = this.soClient.createPointInTimeFinder<Pick<StoredSLO, 'id' | 'revision'>>({
+        type: SO_SLO_TYPE,
+        perPage: 1000,
+        fields: ['id', 'revision'],
+      });
+      let searchAfterKey: SortResults | undefined;
+      let soIdsToCheck: string[] = [];
+      let sloSummaryIdsToDelete: string[] = [];
+
+      for await (const response of finder.find()) {
+        const soItems = response.saved_objects.map(
+          (so) => `${so.attributes.id}${SEPARATOR}${so.attributes.revision}`
+        );
+
+        const { sloSummaryIds, searchAfter } = await this.fetchSloSummariesIds(searchAfterKey);
```

Comment:
One small concern I have with this es search is that we might retrieve a new tuple [slo.id, slo.revision] that was just created after the PIT query to the saved object.
This means that the saved object repository would not have this tuple [slo.id, slo.revision] but it would be returned from the summary documents search, resulting in its deletion.

That's very much a edge case, and the duration between the saved object PIT query and the few summary document requests should be pretty low. Also, the summary transform should recreate the document when it runs next.
I think that's ok to keep as is 🤔  what do you think?